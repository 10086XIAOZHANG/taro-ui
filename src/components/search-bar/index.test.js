import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'

import AtSearchBar from '../../../.temp/components/search-bar/index'

describe('AtSearchBar Snap', () => {
  it('render initial AtSwitch', () => {
    const componet = renderToString(<AtSearchBar />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props value', () => {
    const componet = renderToString(<AtSearchBar value='value' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props placeholder', () => {
    const componet = renderToString(<AtSearchBar placeholder='placeholder' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props maxlength', () => {
    const componet = renderToString(<AtSearchBar maxlength={120} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props disabled', () => {
    const componet1 = renderToString(<AtSearchBar fixed />)
    expect(componet1).toMatchSnapshot()
  })

  it('render AtSearchBar -- props focus', () => {
    const componet1 = renderToString(<AtSearchBar focus />)
    expect(componet1).toMatchSnapshot()
  })

  it('render AtSearchBar -- props disabled', () => {
    const componet1 = renderToString(<AtSearchBar disabled />)
    expect(componet1).toMatchSnapshot()
  })

  it('render AtSearchBar -- props showActionButton', () => {
    const componet1 = renderToString(<AtSearchBar showActionButton />)
    expect(componet1).toMatchSnapshot()
  })

  it('render AtSearchBar -- props actionName', () => {
    const componet1 = renderToString(<AtSearchBar actionName='actionName' />)
    expect(componet1).toMatchSnapshot()
  })
})
