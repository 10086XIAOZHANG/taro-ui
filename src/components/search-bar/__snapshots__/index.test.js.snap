// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtSearchBar Snap render AtSearchBar -- props actionName 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__container\\"><div style=\\"width:100%;\\" class=\\"at-search-bar__placeholder_wrap\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility: visible;\\" class=\\"taro-text at-search-bar__placeholder\\"></span></div><input value=\\"\\" maxlength=\\"120\\" class=\\"weui-input at-search-bar__input\\" type=\\"search\\"/><div style=\\"display: none;\\" class=\\"at-search-bar__clear\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-147px;\\" class=\\"at-search-bar__action\\">actionName</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props disabled 1`] = `"<div class=\\"at-search-bar at-search-bar--fixed\\"><div class=\\"at-search-bar__container\\"><div style=\\"width:100%;\\" class=\\"at-search-bar__placeholder_wrap\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility: visible;\\" class=\\"taro-text at-search-bar__placeholder\\"></span></div><input value=\\"\\" maxlength=\\"120\\" class=\\"weui-input at-search-bar__input\\" type=\\"search\\"/><div style=\\"display: none;\\" class=\\"at-search-bar__clear\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-35px;\\" class=\\"at-search-bar__action\\">取消</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props disabled 2`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__container\\"><div style=\\"width:100%;\\" class=\\"at-search-bar__placeholder_wrap\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility: visible;\\" class=\\"taro-text at-search-bar__placeholder\\"></span></div><input value=\\"\\" maxlength=\\"120\\" class=\\"weui-input at-search-bar__input\\" disabled type=\\"search\\"/><div style=\\"display: none;\\" class=\\"at-search-bar__clear\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-35px;\\" class=\\"at-search-bar__action\\">取消</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props focus 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__container\\"><div style=\\"width:28px;\\" class=\\"at-search-bar__placeholder_wrap\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility: visible;\\" class=\\"taro-text at-search-bar__placeholder\\"></span></div><input value=\\"\\" focus maxlength=\\"120\\" class=\\"weui-input at-search-bar__input\\" type=\\"search\\"/><div style=\\"display: none;\\" class=\\"at-search-bar__clear\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:1;margin-right:0;\\" class=\\"at-search-bar__action\\">取消</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props maxlength 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__container\\"><div style=\\"width:100%;\\" class=\\"at-search-bar__placeholder_wrap\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility: visible;\\" class=\\"taro-text at-search-bar__placeholder\\"></span></div><input value=\\"\\" maxlength=\\"120\\" class=\\"weui-input at-search-bar__input\\" type=\\"search\\"/><div style=\\"display: none;\\" class=\\"at-search-bar__clear\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-35px;\\" class=\\"at-search-bar__action\\">取消</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props placeholder 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__container\\"><div style=\\"width:100%;\\" class=\\"at-search-bar__placeholder_wrap\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility: visible;\\" class=\\"taro-text at-search-bar__placeholder\\">placeholder</span></div><input value=\\"\\" maxlength=\\"120\\" class=\\"weui-input at-search-bar__input\\" type=\\"search\\"/><div style=\\"display: none;\\" class=\\"at-search-bar__clear\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-35px;\\" class=\\"at-search-bar__action\\">取消</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props showActionButton 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__container\\"><div style=\\"width:100%;\\" class=\\"at-search-bar__placeholder_wrap\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility: visible;\\" class=\\"taro-text at-search-bar__placeholder\\"></span></div><input value=\\"\\" maxlength=\\"120\\" class=\\"weui-input at-search-bar__input\\" type=\\"search\\"/><div style=\\"display: none;\\" class=\\"at-search-bar__clear\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:1;margin-right:0;\\" class=\\"at-search-bar__action\\">取消</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props value 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__container\\"><div style=\\"width:28px;\\" class=\\"at-search-bar__placeholder_wrap\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility: hidden;\\" class=\\"taro-text at-search-bar__placeholder\\"></span></div><input value=\\"value\\" maxlength=\\"120\\" class=\\"weui-input at-search-bar__input\\" type=\\"search\\"/><div style=\\"display: flex;\\" class=\\"at-search-bar__clear\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:1;margin-right:0;\\" class=\\"at-search-bar__action\\">取消</div></div>"`;

exports[`AtSearchBar Snap render initial AtSwitch 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__container\\"><div style=\\"width:100%;\\" class=\\"at-search-bar__placeholder_wrap\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility: visible;\\" class=\\"taro-text at-search-bar__placeholder\\"></span></div><input value=\\"\\" maxlength=\\"120\\" class=\\"weui-input at-search-bar__input\\" type=\\"search\\"/><div style=\\"display: none;\\" class=\\"at-search-bar__clear\\"><span style=\\"font-size:15px;color:#999;\\" class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-35px;\\" class=\\"at-search-bar__action\\">取消</div></div>"`;
